api/machines.go

```diff
@@ -27,6 +30,10 @@ import (
 	"github.com/coreos/fleet/schema"
 )
 
+var (
+	metadataPathRegex = regexp.MustCompile("^/([0-9a-f]{32})/metadata/([A-Za-z0-9_.-]+$)")
```

Comment:
fleet's machines don't necessarily have to use systemd's machine-id concept to identify themselves. It can be overridden in config. This needs to be an arbitrary string.